tests/js/sentry-test/reactTestingLibrary.tsx

```diff
@@ -156,18 +203,54 @@ function makeAllTheProviders(options: ProviderOptions) {
  *
  * If your component requires additional context you can pass it in the
  * options.
+ *
+ * To test route changes, pass `disableRouterMocks: true`. This will return a
+ * `router` property which can be used to access the location or manually
+ * navigate to a route.
  */
-function render(
+function render<T extends boolean = false>(
   ui: React.ReactElement,
-  {router, organization, disableRouterMocks, ...rtlOptions}: Options = {}
-) {
+  {
+    router: incomingRouter,
+    organization: incomingOrganization,
+    disableRouterMocks,
+    ...rtlOptions
+  }: Options<T> = {}
+): RenderReturn<T> {
+  const history = createMemoryHistory({
+    initialEntries: [incomingRouter?.location?.pathname ?? LocationFixture().pathname],
+  });
+
   const AllTheProviders = makeAllTheProviders({
-    organization,
-    router,
+    organization: incomingOrganization,
+    router: incomingRouter,
     disableRouterMocks,
+    history,
   });
 
-  return rtl.render(ui, {wrapper: AllTheProviders, ...rtlOptions});
+  const memoryRouter = makeRouter({
+    children: <AllTheProviders>{ui}</AllTheProviders>,
```

Comment:
No longer passing the providers in to RTL's `wrapper` option. Because we need to create the router up front (in order to return it from `render`), we cannot create it in the wrapper function.